src/tests/universal-nrp-e2e/StartLocalTest.sh

```diff
@@ -0,0 +1,288 @@
+#!/bin/bash
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# StartLocalTest.sh
+# Description: This script orchestrates tests on a local machine. Installs dependencies, 
+#              runs tests, and collects logs/reports. Returns an error code if any stage fails.
+#
+# Usage: ./StartLocalTest.sh [-s stage-name] [-p policy-package.zip -c resource-count [-r] [-g]]
+#        -s stage-name:         Specify the stage name. Valid options are: dependency_check, run_tests, collect_logs.
+#                               If no stage is specified, all stages will be executed in this order:
+#                                   dependency_check, run_tests, collect_logs
+#        -p policy-package.zip: The Azure Policy Package to test
+#        -c resource-count:     The number of resources to validate, tests will fail if this doesn't match (Default: 0)
+#        -r remediate-flag:     When the flag is enabled, performs remediation on the Policy Package (Default: No remediation performed)
+#        -g generalize-flag:    Generalize the current machine for tests. Performs the following:
+#                                   - Remove logs and tmp directories
+#                                   - Clean package management cache
+#                                   - Clean cloud-init flags to reset cloud-init to initial-state
+
+powershell_version="7.4.6"
```

Comment:
Is this specific version required/could an already installed version be newer/older and still work (min required version to work)?

Reply:
this is meant to be the version that will be installed, if its already present we just make use of the current without installing.